cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -398,6 +403,110 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	return nil
 }
 
+func (a *StaticAutoscaler) handleQuotaAndStockoutErrors(currentTime time.Time) {
+	currentInstancesPerNodeGroup := a.clusterStateRegistry.GetCloudProviderNodeInstances()
+	previousInstancesPerNodeGroup := a.clusterStateRegistry.GetPreviousCloudProviderNodeInstances()
+	nodeGroups := a.CloudProvider.NodeGroups()
+
+	for _, nodeGroup := range nodeGroups {
+		a.handleQuotaAndStockoutErrorsForNodeGroup(
+			nodeGroup,
+			currentInstancesPerNodeGroup[nodeGroup.Id()],
+			previousInstancesPerNodeGroup[nodeGroup.Id()],
+			currentTime)
+	}
+}
+
+func (a *StaticAutoscaler) handleQuotaAndStockoutErrorsForNodeGroup(
+	nodeGroup cloudprovider.NodeGroup,
+	currentInstances []cloudprovider.Instance,
+	previousInstances []cloudprovider.Instance,
+	currentTime time.Time) {
+
+	_, currentErrorCodeToInstance := a.buildInstanceToOutOfResourcesErrorCodeMappings(currentInstances)
+	previousInstanceToErrorCode, _ := a.buildInstanceToOutOfResourcesErrorCodeMappings(previousInstances)
+
+	// If node group is scaling up and there are new node-create requests which cannot be satisfied because of
```

Comment:
Why only do this for out-of-resources errors? Do we expect other errors to be recoverable?